File: src/arch/aarch64/asm.zig
pub inline fn nop() void {
    asm volatile ("nop");
}

pub inline fn wfe() void {
    asm volatile ("wfe");
}

pub inline fn udf(imm: comptime_int) void {
    asm volatile (
        \\ udf #{imm}
        :
        : [imm] "{imm}" (imm),
    );
}

File: src/arch/aarch64/asm.rs
use core::arch::asm;

#[inline]
pub fn nop() {
    unsafe {
        asm!("nop", options(nostack, nomem));
    }
}

#[inline]
pub fn wfe() {
    unsafe {
        asm!("wfe", options(nostack, nomem));
    }
}

pub fn udf<const IMM: usize>() {
    unsafe {
        asm!("udf #{imm}", imm = const IMM, options(nostack, nomem));
    }
}
